From: Stefan Haberland <sth@linux.vnet.ibm.com>
Subject: dasd: add kmsg descriptions for channel path aware error recovery
Patch-mainline: Never, kmsg in sles only
References: bsc#1025481, FATE#321528, LTC#146543

Summary:     s390/dasd: channel path aware error recovery
Description: With this feature, the DASD device driver more robustly handles 
             DASDs that are attached via multiple channel paths and are
             subject to constant Interface-Control-Checks (IFCCs) and
             Channel-Control-Checks (CCCs) or loss of High-Performance-FICON 
             (HPF) functionality on one or more of these paths.
             If a channel path does not work correctly, it is removed from
             normal operation as long as other channel paths are available.
             All extended error recovery states can be queried and reset via
             user space interfaces.

Upstream-Description:

             dasd: add kmsg descriptions for channel path aware error recovery

Signed-off-by: Stefan Haberland <sth@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.com>
---
 Documentation/kmsg/s390/dasd-eckd |   53 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 53 insertions(+)

--- a/Documentation/kmsg/s390/dasd-eckd
+++ b/Documentation/kmsg/s390/dasd-eckd
@@ -2098,3 +2098,56 @@
  * User action:
  * None.
  */
+
+/*?
+ * Text: "%s: High Performance FICON disabled\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the DASD
+ * Description:
+ * High Performance FICON (HPF) has been disabled. Either the device
+ * lost HPF functionality, or none of the remaining channel paths are
+ * HPF capable.
+ * User action:
+ * Report the problem to your support organization.
+ * Ensure that the cabling between the storage server and the mainframe
+ * system is securely in place.
+ * Reset the device and channel paths by writing "all" or a logical path mask
+ * to the path_reset sysfs attribute of the device.
+ */
+
+/*?
+ * Text: "%s: Channel path %02X lost HPF functionality and is disabled\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the DASD
+ *   @2: logical path mask
+ * Description:
+ * A channel path has lost High Performance FICON (HPF) functionality
+ * and was removed from regular operations.
+ * User action:
+ * Report the problem to your support organization.
+ * Ensure that the cabling between the storage server and the mainframe
+ * system is securely in place.
+ * Reset the device and channel paths by writing "all" or a logical path mask
+ * to the path_reset sysfs attribute of the device.
+ */
+
+/*?
+ * Text: "%s: Path %x.%02x (pathmask %02x) is disabled - IFCC threshold exceeded\n"
+ * Severity: Error
+ * Parameter:
+ *   @1: bus ID of the DASD
+ *   @2: cssid
+ *   @3: chpid
+ *   @4: logical path mask
+ * Description:
+ * Due to numerous interface or channel control checks (IFCCs), a channel path
+ * was removed from regular operations to retain good I/O performance.
+ * User action:
+ * Ensure that the cabling between the storage server and the mainframe
+ * system is securely in place.
+ * Reset the device and channel paths by writing "all" or a logical path mask
+ * to the path_reset sysfs attribute of the device.
+ * If the problem persists, report it to your support organization.
+ */
